### ISSUE TITLE
Invite only Page still accessible via search and via url

### ISSUE THREAD

**Issue by rodgerholl** [moxi9/phpfox#869]
PHPFox 4.1.1.  Any browser
Create an Invite only page.

From an account that is not invited to join the page,  if they search for the page (or a portion of the page name) they see the invite only page in the list of results and can click on it to access the invite only page.  My thoughts were that invite only would restrict access to only those who had been invited.

I am thinking this is an issue.

Trying to create private areas for only authorized accounts to access.   Invite only pages were the only thing I found in V4 that I thought would accomplish the goal.

**Comment by rodgerholl** [comment]
This is basically a fresh install of 4.1.1.  with minor wisteria theme modification.

**Comment by Data66PF** [comment]
It has always worked like this as the search doesn't have a filter like that nor does the direct url. However, in case we look at improving this, to test:

1. Create a front end Group, not Page, as only groups have an option to set permissions as invite only.
2. Set it to invite only for permissions and members only for all the other permission settings.
3. Go use the site search and search for that title of the group and it will show in the search listings, however, you can't actually see any of the content of it which is working as designed and appropriate.
4. Using a direct url to the group, put that in a different browser while logged in and you will see the group but again, the content is restricted as you are not a member.

This enhancement appears to be to restrict the group from showing up at all in any listings and to make it so that the group's direct url is not accessible.

**Comment by rodgerholl** [comment]
I guess this may be terminology confusion?  The menu item to create a front end group is it called a page?   That is what I created.

  I see that with the permissions set they can see the page/group in search,  can get to the url,  but they cant see the content.   which will work for me for now.   

  I noticed that in the PF.Base/module.search/include/service/search.class.php  Query function  (which I believe the is the function that gathers the search results) there is a loop that looks for a callback "filterSearchResults" function in the module's class.
  Was wondering 2 things,

1. If a module has a filterSearchResults function,  it could filter the search results so that these pages / group do not show up in the search.  

2. If so... Is there a way to extent a module (plugin or something) so that the module would have a filterSearchResults function or would we have to modify the Base module .  (not preferred of course)

The forum module has a callback class that contains a filterSearchResults function,  but I have not completely figured out how it filters results yet..
